src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,40 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.service.producers.BulldozerProducer;
+import core.mate.academy.service.producers.ExcavatorProducer;
+import core.mate.academy.service.producers.TruckProducer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        MachineProducer<? extends Machine> producer = null;
```

Comment:
i think this can be just a wildcard or a `Machine` type, usually when you use PECS, you are declaring some methods or classes, but the variable types are getting it from where you are getting an instance of that type, and does not require using `extends`/`super`
```suggestion
        MachineProducer<Machine> producer = null;
```
or you can just return the result right away to avoid creating this variable at all